warehouse/admin/views/users.py

```diff
@@ -125,3 +126,23 @@ def user_detail(request):
         return HTTPSeeOther(location=request.current_route_path())
 
     return {"user": user, "form": form, "roles": roles}
+
+
+@view_config(
+    route_name='admin.user.delete',
+    require_methods=['POST'],
+    permission='admin',
+    uses_session=True,
+    require_csrf=True,
+)
+def user_delete(request):
+    user = request.db.query(User).get(request.matchdict['user_id'])
+
+    # Delete projects one by one so they are purged from the cache
+    for project in user.projects:
+        remove_project(project, request, flash=False)
```

Comment:
Need a `JournalEntry`

Reply:
There's already a `JournalEntry` happening in `remove_project`.

Reply:
confirmed, sorry I tested the button locally on a user with no projects :)